src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocketsHttpHandler.cs

```diff
@@ -345,7 +345,25 @@ private HttpMessageHandler SetupHandlerChain()
                 return Task.FromException<HttpResponseMessage>(error);
             }
 
-            return handler.SendAsync(request, cancellationToken);
+            if (HttpTelemetry.IsEnabled && request.RequestUri != null)
+            {
+                // Wrap the task for event activity-id threading.
+                HttpTelemetry.Log.RequestStart(request.RequestUri.IdnHost, request.RequestUri.Port);
+                Task<HttpResponseMessage> task = handler.SendAsync(request, cancellationToken);
+                task.ContinueWith(
+                    t => HttpTelemetry.Log.RequestAbort(),
+                    cancellationToken,
+                    TaskContinuationOptions.OnlyOnCanceled | TaskContinuationOptions.OnlyOnFaulted | TaskContinuationOptions.ExecuteSynchronously, TaskScheduler.Default);
+                task.ContinueWith(
+                    t => HttpTelemetry.Log.RequestStop(request.RequestUri.IdnHost, request.RequestUri.Port),
+                    cancellationToken,
+                    TaskContinuationOptions.ExecuteSynchronously, TaskScheduler.Default);
```

Comment:
The cancellation token provided to `ContinueWith` will cancel the continuation from running, which will cause the initial `RequestStart` to not see a corresponding end.

Also consider passing in the `request` in for `Object`-taking overload, to prevent allocating a per-request closure.

It might also be faster to return the continuation task instead of the original task. I believe @stephentoub made a change that allows only a single continuation to run synchronously, so when the user `await`s the original `Task` it will always end up punted to the scheduler due to your continuation getting synchronous priority.

Reply:
We should combine both continuations into a single async method, e.g.:
```C#
return HttpTelemetry.IsEnabled && request.RequestUri != null ?
    WithLogging(handler, request, cancellationToken) :
    handler.SendAsync(request, cancellationToken);

static async Task WithLogging(HttpMessageHandler handler, HttpRequestMessage request, CancellationToken cancellationToken)
{
    HttpTelemetry.Log.RequestStart(request.RequestUri.IdnHost, request.RequestUri.Port);
    try
    {
        await handler.SendAsync(request, cancellationToken).ConfigureAwait(false);
    }
    catch
    {
        HttpTelemetry.Log.RequestAbort(); // this doesn't need the request passed in?
        throw;
    }
    finally
    {
        HttpTelemetry.Log.RequestStop(request.RequestUri.IdnHost, request.RequestUri.Port);
    }
}
```